### PR TITLE
Matrix conversion method

### PR DESCRIPTION
**Pull request by kescobo**
Addresses #1474.

There are a few failing tests and a bunch of deprecation warnings, but a quick glance suggests that they're the same as on Master.

Please let me know if there's anything else I can do.

cc @nalimilan @bkamins

### CODE REVIEW

**Comment by kescobo** [comment]
Odd - it looks like my text editor got cute and removed a bunch of spaces at the end of lines. let me know if you'd like me to strip out those changes.

File: src/DataFrames.jl [hunk 1]
```diff
@@ -8,7 +8,7 @@ module DataFrames
 
 using Reexport, StatsBase, SortingAlgorithms, Compat, Statistics, Unicode, Printf
 @reexport using CategoricalArrays, Missings
-using Base.Sort, Base.Order
+using Base.Sort, Base.Order, Base.Matrix
```
Comment (bkamins) [inline src/DataFrames.jl]: I do not think you need this line

File: src/DataFrames.jl [hunk 2]
```diff
@@ -57,6 +57,7 @@ export AbstractDataFrame,
        head,
        tail,
        permutecols!,
+       Matrix,
```
Comment (bkamins) [inline src/DataFrames.jl]: Similar with this line.

**Comment by bkamins** [comment]
It would probably be better to split removal of the spaces to another commit but it is good that it is cleaned up 😄.

File: src/DataFrames.jl (hunk 1 above)
Comment (kescobo) [reply src/DataFrames.jl]: Right you are.

**Comment by kescobo** [comment]
@bkamins is there an easy way to go back and move it to a new commit? I can start over and just blow away the current commits (not like there's a lot of work involved)

**Comment by bkamins** [comment]
You can (I am not a git wizard but this is what I do in such cases):
* revert all 3 commits in git in your local repo using `git reset` - this will still leave them in your working directory (alternatively if you add `--hard` you will remove all changes from your working directory and will have to rewrite the change anew)
* then make two commits - one for the change and one for stripping endline white spaces (here you might need to commit selected lines in a file - git allows for this)
* `git push --force`

But I guess we can leave it as is as if you have problems with this as it is not hugely confusing.

**Comment by kescobo** [comment]
No, I've done stuff like that before, just wasn't sure how you felt about rewriting history :-). Will get that done now.

**Comment by kescobo** [comment]
I think this should be good to go, unless you want me to rebase off of the latest master.

**Comment by nalimilan** [comment]
Thanks!

**Comment by kescobo** [comment]
My pleasure! Thanks for speedy review

**Pull request merged**